Lib/typing.py

```diff
@@ -2925,6 +2927,43 @@ def __round__(self, ndigits: int = 0) -> T:
         pass
 
 
+@runtime_checkable
```

Comment:
What is the reasoning for making it runtime checkable?

Reply:
Considering that readable classes often have subtly different signatures of `read()` and `readline()` methods (e.g. many do not have a `size` parameter), making this one in particular runtime-checkable does seem a little dangerous. The runtime-checkable machinery just checks that a method on the other class _exists_; it doesn't check the signature on the other class at all

Reply:
So that it's possible to do things like that:

```python
def read_it(s: str | Reader):
    if isinstance(s, str):
        ...
    elif not isinstance(s, Reader):
        raise TypeError(...)

    ...
```

I'm not terribly in invested in this, but is there a reason not to mark it as `@runtime_checkable`?

Reply:
I think Alex answered this one for you :)